optaplanner-examples/src/main/java/org/optaplanner/examples/tennis/optional/score/TennisConstraintProvider.java

```diff
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.examples.tennis.optional.score;
+
+import static org.optaplanner.core.api.score.stream.Joiners.equal;
+import static org.optaplanner.core.api.score.stream.Joiners.lessThan;
+
+import java.util.function.BiFunction;
+import java.util.function.Function;
+
+import org.apache.commons.lang3.tuple.Pair;
+import org.optaplanner.core.api.score.buildin.hardmediumsoft.HardMediumSoftScore;
+import org.optaplanner.core.api.score.stream.Constraint;
+import org.optaplanner.core.api.score.stream.ConstraintFactory;
+import org.optaplanner.core.api.score.stream.ConstraintProvider;
+import org.optaplanner.core.impl.score.stream.bi.DefaultBiConstraintCollector;
+import org.optaplanner.core.impl.score.stream.uni.DefaultUniConstraintCollector;
+import org.optaplanner.examples.common.solver.drools.functions.LoadBalanceByCountAccumulateFunction;
+import org.optaplanner.examples.common.solver.drools.functions.LoadBalanceByCountAccumulateFunction.LoadBalanceByCountData;
+import org.optaplanner.examples.common.solver.drools.functions.LoadBalanceByCountAccumulateFunction.LoadBalanceByCountResult;
+import org.optaplanner.examples.tennis.domain.TeamAssignment;
+import org.optaplanner.examples.tennis.domain.UnavailabilityPenalty;
+
+public final class TennisConstraintProvider implements ConstraintProvider {
+
+    private static final String CONSTRAINT_PACKAGE = "org.optaplanner.examples.tennis.solver";
```

Comment:
Why do we need that for the tennis example, but not for CloudBalancingConstraintProvider which is also in the "optional.score" package?

Reply:
Any way to get rid of having to pass the constraintPackage to the penalize methods?

Reply:
Yes, two options:

- Adapt both this and the DRL to `ConstraintConfiguration`.
- Relax the requirement that `FULL_ASSERT` must pass for both, which it won't, when the packages differ.